mobile/mysterium/wireguard_connection_setup.go

```diff
@@ -157,143 +133,208 @@ func newTunnDevice(wgTunnSetup WireguardTunnelSetup, config *wireguard.ServiceCo
 		time.Sleep(time.Duration(config.Consumer.ConnectDelay) * time.Millisecond)
 	}
 
+	// Wait for local port to become available since it will be used as WireGuard listen port
+	// when provider is behind NAT.
+	if config.LocalPort > 0 {
+		log.Info().Msgf("Waiting for port %d to become available", config.LocalPort)
+		if err := waitUDPPortReadyFor(config.LocalPort, 10*time.Second); err != nil {
+			return nil, errors.Wrap(err, "failed to wait for UDP port")
+		}
+	}
+
 	fd, err := wgTunnSetup.Establish()
 	if err != nil {
 		return nil, err
 	}
 	log.Info().Msgf("Tun value is: %d", fd)
-	tun, err := newDeviceFromFd(fd)
+	tunDevice, err := newDeviceFromFd(fd)
 	if err == nil {
 		//non-fatal
-		name, nameErr := tun.Name()
+		name, nameErr := tunDevice.Name()
 		log.Info().Err(nameErr).Msg("Name value: " + name)
 	}
 
-	return tun, err
+	return tunDevice, err
+}
+
+func waitUDPPortReadyFor(port int, timeout time.Duration) error {
+	timeoutChan := time.After(timeout)
```

Comment:
This can be smaller, not sure if it's better, but smaller. 😂
```golang
done := time.Now().Add(timeout)
for t := time.NewTicker(500 * time.Millisecond); time.Now().Before(done); <-t.C {
	p, err := net.ListenPacket("udp", fmt.Sprintf(":%d", port))
	if err == nil {
		p.Close()
		return nil
	} 
	log.Err(err).Msgf("Port %d is in use. Trying to check again...", port)
}
return fmt.Errorf("timeout waiting for UDP port %d", port)
```